crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_none_literal.rs

```diff
@@ -87,8 +90,34 @@ pub(crate) fn redundant_none_literal<'a>(checker: &mut Checker, literal_expr: &'
     let fix = if other_literal_elements_seen {
         None
     } else {
+        // Avoid producing syntax errors when `Literal[None] | None` would be fixed to
+        // `None | None`. Instead fix to `None`. No action needed from `typing.Union`,
+        // as `Union[None, None]` is valid syntax.
```

Comment:
`None | None` is invalid _Python_, but it's valid _syntax_ (the code compiles, it just raises an exception!)

```suggestion
        // as `Union[None, None]` is valid Python.
```